### ISSUE TITLE
Possible Bug in Master

### ISSUE THREAD

**Issue by jainsamyak** [jcmturner/gokrb5#472]
Hello,
We noticed an issue with this library where due to a recent [commit](https://github.com/jcmturner/gokrb5/commit/ce611a4fb5015653f67f163ec8ccfb162035eded), the builds started breaking. It seems that the function trying to be used (io.ReadFile) does not exist in the package at all. We noticed this issue as our project relies on [Sarama](https://github.com/Shopify/sarama), which is internally using this library for Golang.
Kindly help provide a resolution.
Details:
PR: https://github.com/jcmturner/gokrb5/pull/463
Commit: ce611a4fb5015653f67f163ec8ccfb162035eded
Precise line references:
- https://github.com/jcmturner/gokrb5/commit/ce611a4fb5015653f67f163ec8ccfb162035eded#diff-e82a52c36d5d14a1dd213ae90b7cfee76368eacfde38aa0419922ec8f8ff4768R173
- https://github.com/jcmturner/gokrb5/commit/ce611a4fb5015653f67f163ec8ccfb162035eded#diff-15fdd9e522e45a4261321a49041838ff4470d5580b97d124d2afa8db25f88f1fR66

**Note:** This seems to be an issue specific to the master branch, as when we switched to the release version `v8.4.2`, everything went back to normal.

Thanks & Regards.

**Screenshots for Ref:**
<img width="623" alt="image (9)" src="https://user-images.githubusercontent.com/33225811/176986780-14782c4b-e08f-4283-a7ba-a7505fce9f45.png">
<img width="794" alt="image (10)" src="https://user-images.githubusercontent.com/33225811/176986782-45ec0f2e-d4d4-4ec2-b9fa-3d34810d2b68.png">

**Comment by marcoantoniobferreira** [comment]
Hey @jainsamyak! I'm having the same issue, but I didn't get it how to switch branchs, because I'm using Sarama, and they have `gokrb` as dependency. Even when i tried to upgrade Sarama to the latest version `v1.34.1`, the `gokrb` was also updated automatically to `v8.4.2`, but the issue goes on. Could you provide i little more detailed feedback on how you fixed it? Thank you!
<img width="565" alt="Screen Shot 2022-07-12 at 13 39 16" src="https://user-images.githubusercontent.com/23589475/178546718-41bc333b-88cc-4c3c-8b3f-1a1c07d6ab09.png">

**Comment by jainsamyak** [comment]
Hi @marcoantoniobferreira ,
I simply added a dependency to the release version of gokrb5 in our glide.yaml file, try and set a manual dependency for the library to [v8.4.2](https://github.com/jcmturner/gokrb5/tree/v8.4.2)

**Comment by marcoantoniobferreira** [comment]
I see! Thank you @jainsamyak!

**Comment by jcmturner** [comment]
I believe that os.ReadFile was introduced in Go 1.16 (https://go.dev/doc/go1.16) as part of deprecating the io/ioutil. Are you using an older version of Go? The master branch will be the basis of a future released version so I recommend updating the version of Go you are using if this is the case. As this does not seem to be an issue with the gokrb5 library I'll be closing this issue. Thanks.

**Comment by marcoantoniobferreira** [comment]
Yes @jcmturner, i am using Go 1.13. I will update to 1.16 and see if works. Thank you!
